packages/algorand-js/src/runtime/runtime.ts

```diff
@@ -0,0 +1,215 @@
+   * @param appId: current application id
+   * @param key: key to fetch value of from local state
+   * @param value: key to fetch value of from local state
+   */
+  updateGlobalState (appId: number, key: Uint8Array, value: StackElem): SSCStateSchema[] {
+    const appDelta = this.assertAppDefined(appId);
+    const globalState = appDelta["global-state"];
+
+    const data = getKeyValPair(key, value); // key value pair to put
+    const idx = globalState.findIndex(schema => compareArray(schema.key, key));
+    if (idx === -1) {
+      globalState.push(data); // push new pair if key not found
+    } else {
+      globalState[idx].value = data.value; // update value if key found
+    }
+    appDelta["global-state"] = globalState; // save updated state
+
+    assertValidSchema(appDelta["global-state"], appDelta["global-state-schema"]); // verify if updated schema is valid by config
+    return globalState;
+  }
+
+  /**
+   * Description: set accounts for context as {address: SDKAccount}
+   * @param accounts: array of account info's
+   */
+  createStatefulContext (accounts: SDKAccount[]): void {
+    for (const acc of accounts) {
+      this.store.accounts.set(acc.address, acc);
+
+      for (const app of acc.createdApps) {
+        this.store.globalApps.set(app.id, app.params);
+      }
+    }
+  }
+
+  /**
+   * Description: creates a new transaction object from given execParams
+   * @param txnParams : Transaction parameters for current txn or txn Group
+   * @returns: [current transaction, transaction group]
+   */
+  createTxnContext (txnParams: ExecParams | ExecParams[]): [Txn, Txn[]] {
+    // if txnParams is array, then user is requesting for a group txn
+    if (Array.isArray(txnParams)) {
+      if (txnParams.length > 16) {
+        throw new Error("Maximum size of an atomic transfer group is 16");
+      }
+
+      const txns = [];
+      for (const txnParam of txnParams) { // create encoded_obj for each txn in group
+        const mockParams = mockSuggestedParams(txnParam.payFlags);
+        const tx = mkTransaction(txnParam, mockParams);
+
+        // convert to encoded obj for compatibility
+        const encodedTxnObj = tx.get_obj_for_encoding() as Txn;
+        encodedTxnObj.txID = tx.txID();
+        txns.push(encodedTxnObj);
+      }
+      assignGroupID(txns); // assign unique groupID to all transactions in the array/group
+      return [txns[0], txns]; // by default current txn is the first txn (hence txns[0])
+    } else {
+      // if not array, then create a single transaction
+      const mockParams = mockSuggestedParams(txnParams.payFlags);
+      const tx = mkTransaction(txnParams, mockParams);
+
+      const encodedTxnObj = tx.get_obj_for_encoding() as Txn;
+      encodedTxnObj.txID = tx.txID();
+      return [encodedTxnObj, [encodedTxnObj]];
+    }
+  }
+
+  prepareInitialState (txnParams: ExecParams| ExecParams[], accounts: SDKAccount[]): Context {
```

Comment:
Now I see what you want to do. It's not Runtime setup, it's Tx context creation.
1. The way you create copy is wrong
2. Please rename this function, eg: `createTxContext`

Reply:
I have removed this function.